python/arcticdb/version_store/_store.py

```diff
@@ -284,12 +292,14 @@ def __setstate__(self, state):
         custom_norm.__setstate__(state["custom_norm"])
         env = state["env"]
         open_mode = state["open_mode"]
+        credential = state["credential"]
```

Comment:
Bit worried about the pickling stuff. Can we leave it as none or something and reconstruct in the normal case (where no special credential is set).

IIRC all the pickling stuff is in aid of Python forking. Have you tested (manually) that this procedure works correctly when using Azure AD?

It's also a bit worrying since the `Credential` is basically `Azure::...::TokenCredential` and we presumably don't have any guarantees of its format.